File: templates/csharp/app-mvvm/ViewModels/MainWindowViewModel.cs
﻿namespace AvaloniaAppTemplate.ViewModels;

#if (CommunityToolkitChosen)
public partial class MainWindowViewModel : ViewModelBase
#else
public class MainWindowViewModel : ViewModelBase
#endif
{
    public string Greeting { get; } = "Welcome to Avalonia!";
}

File: templates/fsharp/app-mvvm/ViewModels/MainWindowViewModel.fs
namespace AvaloniaAppTemplate.ViewModels

#if (CommunityToolkitChosen)
open CommunityToolkit.Mvvm.ComponentModel;
#endif

type MainWindowViewModel() =
#if (CommunityToolkitChosen)
    inherit ObservableObject()
#else
    inherit ViewModelBase()
#endif

    member this.Greeting = "Welcome to Avalonia!"
